src/ORM/Table.php

```diff
@@ -842,8 +843,9 @@ public function behaviors(): BehaviorRegistry
     /**
      * Get a behavior from the registry.
      *
-     * @param string $name The behavior alias to get from the registry.
-     * @return \Cake\ORM\Behavior
+     * @template TName of key-of<TBehaviors>
+     * @param TName $name The behavior alias to get from the registry.
+     * @return TBehaviors[TName]
```

Comment:
wouldnt it be better to keep the param and return for normal IDEs to understand this
And only add phstan specific one on top (phpstan-param?) for the tool to understand it?

Reply:
Update code to use phpstan 'prefix'.